Sources/AWSSDKSwiftCore/AWSClient.swift

```diff
@@ -122,13 +122,7 @@ extension AWSClient {
     }
 
     private func createHTTPClient(for nioRequest: Request) -> HTTPClient {
-        let client: HTTPClient
-        if let _ = self._endpoint {
-            client = HTTPClient(hostname: nioRequest.head.host!, port: nioRequest.head.port ?? 443)
-        } else {
-            client = HTTPClient(hostname: nioRequest.head.hostWithPort!, port: 443)
-        }
-        return client
+        return HTTPClient(hostname: nioRequest.head.hostWithPort!, port: nioRequest.head.port ?? 443)
```

Comment:
Should the default port switch against the scheme? If http should be 80 rather than always assuming 443 for https.

Reply:
As I understand it all aws access is https, so we can assume 443. Ideally we should be extracting all this info from a URL but I wanted to make as small a change as possible because this isn't my code.

Reply:
I understand. It caught my attention because when I tried to workaround this issue, I tried to expose the fake SQS service to port `80` so that `Host: localhost` would match (before this PR), but I was unable to make it work because of this assumption.

Maybe it can be discussed in another pull request?

Reply:
It's certainly an area that could be cleaned